util/pkv_calculation.py

```diff
@@ -0,0 +1,248 @@
+                meter_id, str(date))
+    return None
+
+
+def define_base_values(inhabitants, date):
+    """ Define the base values for a user on a given date.
+    :param int inhabitants: the number of inhabitants in the user's flat
+    :param datetime date: the start date of the calculation which cannot
+    lie in the future
+    :return: the base values for the given meter id and date or None on wrong
+    date parameter
+    :rtype: dict or type(None)
+    """
+
+    # Check input parameter date
+    if check_input_parameter_date(date) is False:
+        logger.info(
+            'The start date of the calculation cannot lie in the future.')
+        return None
+
+    # Define day_zero := Berechnungsstart - 1 Tag
+    day_zero = date - timedelta(days=1)
+
+    # Define consumption := 0
+    consumption = 0.0
+
+    # On day_zero, consumption_cumulated := consumption (kWh)
+    consumption_cumulated = consumption
+
+    # Calculate pkv (Pro-Kopf-Verbrauch) := consumption/inhabitants (kwH)
+    pkv = consumption/inhabitants
+
+    # On day_zero, pkv_cumulated := pkv (kWh)
+    pkv_cumulated = pkv
+
+    # Define days (since calculation start) := 0 (number)
+    days = 0
+
+    # Define moving_average on day_zero := 0.0 (kWh)
+    moving_average = 0.0
+
+    # Calculate moving_average_annualized := moving_average * 365 (kWh, rounded)
+    moving_average_annualized = round(moving_average * 365)
+
+    # Return base values as dict
+    base_values = dict(date=day_zero,
+                       consumption=consumption,
+                       consumption_cumulated=consumption_cumulated,
+                       inhabitants=inhabitants,
+                       pkv=pkv,
+                       pkv_cumulated=pkv_cumulated,
+                       days=days,
+                       moving_average=moving_average,
+                       moving_average_annualized=moving_average_annualized)
+    return base_values
+
+
+def calc_pkv(meter_id, inhabitants, date, session):
+    """ Calculate the Pro-Kopf-Verbrauch for a given user on a given date.
+    :param str meter_id: the user's meter id
+    :param int inhabitants: the number of inhabitants in the user's flat
+    :param datetime.date date: the calculation day which cannot lie in the future
+    :return: the pkv values for the given meter id and date or None if there is
+    an error
+    :rtype: dict or type(None)
+    """
+
+    # Check input parameter date
+    if check_input_parameter_date(date) is False:
+        logger.info(
+            'The input parameter \'date\' cannot lie in the future.')
+        return None
+
+    # Calculate consumption := last meter reading of date - first meter reading
+    # of date in kWh
+
+    consumption_mywh_last = get_last_meter_reading_date(meter_id, date)
+    consumption_mywh_first = get_first_meter_reading_date(meter_id, date)
+
+    if consumption_mywh_last is None or consumption_mywh_first is None:
```

Comment:
Maybe log this absent of data?

Reply:
It gets already logged here: https://github.com/buzzn/mybuzzn-backend/blob/feat-APP-181/tests/test_pkv_calculation.py